src/test/java/org/gridsuite/modification/server/modifications/BatteryModificationTest.java

```diff
@@ -217,6 +218,40 @@ public void testDroopUnchanged() throws Exception {
         assertEquals(18f, createdModification.getDroop().getValue());
     }
 
+    @Test
+    public void testImpactsAfterActivePowerControlModifications() throws Exception {
+        BatteryModificationInfos batteryModificationInfos = (BatteryModificationInfos) buildModification();
+        String modificationToCreateJson = mapper.writeValueAsString(batteryModificationInfos);
+        mockMvc.perform(post(getNetworkModificationUri()).content(modificationToCreateJson).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk()).andReturn();
+        Battery battery = getNetwork().getBattery("v3Battery");
+        assertEquals(0.1f, battery.getExtension(ActivePowerControl.class).getDroop());
+        assertEquals(true, battery.getExtension(ActivePowerControl.class).isParticipate());
+        //modify only droop
+        batteryModificationInfos.setDroop(new AttributeModification<>(0.5f, OperationType.SET));
+        modificationToCreateJson = mapper.writeValueAsString(batteryModificationInfos);
+        MvcResult mvcResult = mockMvc.perform(post(getNetworkModificationUri()).content(modificationToCreateJson).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk()).andReturn();
+        //check impacts
+        String resultAsString = mvcResult.getResponse().getContentAsString();
+        NetworkModificationResult networkModificationResult = mapper.readValue(resultAsString, NetworkModificationResult.class);
+        assertEquals(1, networkModificationResult.getNetworkImpacts().size());
+        assertEquals(1, networkModificationResult.getImpactedSubstationsIds().size());
+        assertEquals("[s2]", networkModificationResult.getImpactedSubstationsIds().toString());
+        //modify only participate
+        batteryModificationInfos.setParticipate(new AttributeModification<>(false, OperationType.SET));
+        modificationToCreateJson = mapper.writeValueAsString(batteryModificationInfos);
+        mvcResult = mockMvc.perform(post(getNetworkModificationUri()).content(modificationToCreateJson).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk()).andReturn();
+        //check impacts
+        resultAsString = mvcResult.getResponse().getContentAsString();
+        networkModificationResult = mapper.readValue(resultAsString, NetworkModificationResult.class);
+        assertEquals(1, networkModificationResult.getNetworkImpacts().size());
+        assertEquals(1, networkModificationResult.getImpactedSubstationsIds().size());
+        assertEquals("[s2]", networkModificationResult.getImpactedSubstationsIds().toString());
+
```

Comment:
Wouldn't it be interesting to also check the battery ActivePowerControl values after the modifications ?

Reply:
this is not the purpose of this PR, the test of modification is done in the other methods

Reply:
Fair enough